tests/test_http_downloader.py

```diff
@@ -0,0 +1,61 @@
+import tempfile
+from unittest.mock import patch
+
+import pytest
+
+from datasetinsights.io.downloader.http_downloader import HTTPDownloader
+from datasetinsights.io.exceptions import ChecksumError
+
+
+def test_read_checksum_from_txt():
+    with tempfile.NamedTemporaryFile(mode="w+") as tmp:
+        tmp.write("123456")
+        tmp.read()
```

Comment:
I think it makes more sense to do a tmp.flush() to force the write to the file instead of tmp.read()